gradle-diff-task-dependencies/plugin/src/main/groovy/com/novoda/gradle/diffdependencytask/DiffDependencyTaskConfig.groovy

```diff
@@ -0,0 +1,11 @@
+package com.novoda.gradle.diffdependencytask
+
+import com.novoda.gradle.diffdependencytask.git.ChangedFilesProvider
+
+class DiffDependencyTaskConfig {
+
+    public GroovyCallable<List<String>> changedFilesProvider = ChangedFilesProvider.create()
+
+    public ConditionalDependencyRepository conditionalDependencyRepository = new MemoryConditionalDependencyRepository()
```

Comment:
We can inject whatever changed files provider and repository in here and test the plugin with mock data and behaviour. Isn't this the only way to configure a plugin?